src/components/movieForm/MovieForm.js

```diff
@@ -18,37 +20,125 @@ Object.keys(formLabels).forEach((key) => {
     formLabels[key] = formLabels[key].toUpperCase();
 });
 
-const MovieForm = ({formTitle, submitCallback, initialMovie, showModal, onClose}) => {
+const validate = values => {
+    const errors = {};
+    if (!values.title) {
+        errors.title = 'Required';
+    } else if (values.title.length > 20) {
+        errors.title = 'Must be 15 characters or less';
+    }
+
+    if (!values.release_date) {
+        errors.release_date = 'Required';
+    } else if (!/(20|19)[0-9]{2}-([0-1][0-9]|[0-2][0-9])-([0-2][0-9]|[3][0-1])/.test(values.release_date)) {
+        errors.release_date = 'Invalid date';
+    }
+
+    if (!values.poster_path) {
+        errors.poster_path = 'Required';
+    } else if(!/https/gm.test(values.poster_path)) errors.poster_path = 'Invalid format'
+
+    if (!values.vote_average) {
+        errors.vote_average = 'Required';
+    } else if (typeof(+values.vote_average) !== 'number') errors.vote_average='Add a number';
+
+    if(!values.runtime) {
+        errors.runtime = 'Required';
+    } else if (typeof(+values.runtime) !== 'number') errors.runtime='Add a number';
+
+    if(!values.overview) errors.overview = 'Required';
+
+    return Object.keys(errors).length === 0 ? null : errors;
+};
+
+const MovieForm = ({formTitle, showModal}) => {
+    const navigate = useNavigate();
+    const {movieId} = useParams();
+    const [movie, setMovie ] = useState();
+    useEffect(() => {
+        let controller, signal;
+        const fetchData = async () => {
+            if (controller) {
+                controller.abort()
+            }
+            controller = new AbortController();
+            signal = controller.signal;
+
+            const response = await fetch(`http://localhost:4000/movies/${movieId}`, {
+                signal: signal,
+            })
+            const data = await response.json();
+            controller = null;
+            setMovie(data);
+        }
+
+        movieId && fetchData();
+    },[movieId]);
     const handleReset = () => {
         document.querySelector('form').reset();
     };
 
-    const handleSubmit = (event) => {
-        event.preventDefault();
-        const formData = Object.fromEntries(new FormData(event.target));
-        submitCallback(formData);
-        return false;
-    };
+    console.warn(movie)
+    const formik = useFormik({
+        initialValues: {
+            title: movie?.title ?? '',
+            release_date: movie?.release_date ?? '',
+            poster_path: movie?.poster_path ?? '',
+            runtime: movie?.runtime ?? null,
+            overview: movie?.overview ?? '',
+            vote_average: movie?.vote_average ?? null,
+        },
+        validate,
+        onSubmit: async (values) => {
+            await fetch('http://localhost:4000/movies', {
+                method: movie?.id ? 'PUT' : 'POST',

```

Comment:
following REST principles, it is better to have `/entity/{id} ` for PUT and `/entity` for POST
https://restfulapi.net/rest-put-vs-post/